jobs/base_model_validator_errors.py

```diff
@@ -0,0 +1,108 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Error classes for model validations."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import cron_services
+import python_utils
+
+PERIOD_TO_HARD_DELETE_MODEL_IN_DAYS = (
+    cron_services.PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED.days)
+
+
+class ModelValidationError(python_utils.OBJECT):
+    """Base error class for model validations."""
+
+    @property
+    def key(self):
+        """Property that returns the error class name."""
+        return self.__class__.__name__
+
+    @property
+    def message(self):
+        """Message property to override in subclasses."""
+        return None
+
+    def __repr__(self):
+        return '%s: %s' % (self.key, self.message) if self.message else self.key
+
+    def __eq__(self, other):
+        if self.__class__ is other.__class__:
+            return (self.key, self.message) == (other.key, other.message)
+        return NotImplemented
```

Comment:
Should this be NotImplementedError?

Reply:
Fixed.

Reply:
`NotImplemented` is correct here, see: https://docs.python.org/3/library/constants.html#NotImplemented

Reply:
Oh -- so sorry, @shubha-rajan! I confused it with NotImplementedError; this is actually the first time I've heard of NotImplemented. Apologies.

Reply:
Feel free to add a comment to the link I provided if you think that'd be helpful.